Add OpenStack Client Config module

emonty opened the pull request:
All of the ansible OpenStack modules are driven by a clouds.yaml config
file which is processed by os-client-config. Expose the data returned by
that library to enable playbooks to iterate over available clouds.

emonty committed:
Add OpenStack Client Config module

All of the ansible OpenStack modules are driven by a clouds.yaml config
file which is processed by os-client-config. Expose the data returned by
that library to enable playbooks to iterate over available clouds.

emonty commented:
@j2sol @Shrews check me on this?

omgjlk commented:
If the purpose of this is to generate facts, perhaps this should just be a facts module and inject that data directly into ansible facts rather than force the user to register the return and process it.

http://docs.ansible.com/developing_modules.html#module-provided-facts

omgjlk commented on cloud/openstack/os_client_config.py:
```diff
@@ -0,0 +1,70 @@
+
+EXAMPLES = '''
+# Inject facts about OpenStack clouds
+- os-client-config
```
This example seems to be ... missing some content.

emonty committed:
Return a list of OpenStack clouds

The main use of this is to feed jinja templating, so structure the
data returned slightly better for that purpose.

omgjlk commented:
I think this looks good now. :+1:

Shrews commented:
I'm having issues running it. Instantiating the AnsibleModule object needs an argspec.

emonty committed:
AnsibleModule takes a dict as a param

Shrews commented:
Works for me now with the latest commit.
